scipy/ndimage/filters.py

```diff
@@ -618,10 +618,18 @@ def _correlate_or_convolve(input, weights, output, mode, cval, origin,
     if not weights.flags.contiguous:
         weights = weights.copy()
     output = _ni_support._get_output(output, input)
+    temp_needed = numpy.shares_memory(input, output)
```

Comment:
This probably should use `may_share_memory`, because `shares_memory` has potentially exponential runtime.

Reply:
Thanks Pauli, good point. Done in gh-12095

Reply:
Thanks for finding and fixing this. I was not aware of the potential exponential runtime.